### PR TITLE
Send weekend calendar on thursday evening

### PR DESCRIPTION
**Pull request by martijnjanssen**

### CODE REVIEW

File: f1_schedule_telegram_bot/main.py [hunk 1]
```diff
@@ -171,6 +171,51 @@ async def fetch_ical() -> Calendar:
...
+    """Send a message with the calendar for the current weekend."""
+    try:
+        cal = await fetch_ical()
+    except requests.exceptions.Timeout as err:
+        logging.warning(f"unable to get iCal: {err}")
+        return
+
+    # Order the events such that the quali is listed before the race
+    events = sorted(cal.events)
+    utcnow = arrow.utcnow().shift(days=-3)
```
Comment (Fastjur) [inline f1_schedule_telegram_bot/main.py:184]: Perhaps a bit of a misleading name if it’s shifted. Fine tho imo.

File: f1_schedule_telegram_bot/main.py [hunk 2]
```diff
@@ -171,6 +171,51 @@ async def fetch_ical() -> Calendar:
...
+    message = ""
+
+    for event in events:
+        # Get the quali and race for this weekend
+        if (
+            utcnow < event.begin
+            and event.begin <= utcnow.shift(days=4)
+            and ("F1: Qualifying" in event.name or "F1: Grand Prix" in event.name)
```
Comment (Fastjur) [inline f1_schedule_telegram_bot/main.py:192]: I think I’d prefer lower casing it all, should prevent some issues if they change case.

File: f1_schedule_telegram_bot/main.py [hunk 3]
```diff
@@ -171,6 +171,51 @@ async def fetch_ical() -> Calendar:
...
+        ):
+            race_name = event.name.split("(")[1].split(")")[0]
+            event_name = event.name.split("F1:")[1].split("(")[0].strip()
+
+            # If message is empty, start with the name of the race
+            if message == "":
+                message += f"<b>{race_name}</b>\n"
```
Comment (Fastjur) [inline f1_schedule_telegram_bot/main.py:199]: Might be nice for a future issue to also do a bit more nice formatting like this for the other updates!

**Review by Fastjur — APPROVED** [review]
Small comments. Fine either way with me! Thanks :)

File: f1_schedule_telegram_bot/main.py (hunk 3 above)
Comment (Fastjur) [reply f1_schedule_telegram_bot/main.py:199]: Created #46

File: f1_schedule_telegram_bot/main.py (hunk 1 above)
Comment (martijnjanssen) [reply f1_schedule_telegram_bot/main.py:184]: Whoops, that's debug

**Review by Fastjur — APPROVED** [review]

**Pull request merged**